server/cluster/cluster.go

```diff
@@ -404,6 +478,37 @@ func (c *Cluster) RegisterNode(ctx context.Context, nodeName string, lease uint3
 	return nil
 }
 
+func (c *Cluster) allocSchemaID(ctx context.Context) (uint32, error) {
+	id, err := c.schemaIDAlloc.Alloc(ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "alloc schema id failed")
+	}
+	return uint32(id), nil
+}
+
+func (c *Cluster) allocTableID(ctx context.Context) (uint64, error) {
+	id, err := c.tableIDAlloc.Alloc(ctx)
+	if err != nil {
+		return 0, errors.Wrap(err, "alloc table id failed")
+	}
+	return id, nil
+}
+
+func (c *Cluster) pickOneShardOnNode(nodeName string) (uint32, error) {
+	if node, ok := c.nodesCache[nodeName]; ok {
+		if len(node.shardIDs) == 0 {
+			return 0, ErrNodeShardsIsEmpty.WithCausef("nodeName:%s", nodeName)
+		}
+
+		id, err := rand.Int(rand.Reader, big.NewInt(int64(len(node.shardIDs))))
```

Comment:
Is it better to generate random number in this way than using `math.rand`, which at least looks more simple.